### PR TITLE
ObjectDetection App

### PR DESCRIPTION
**Pull request by ayermolo**
Summary:
Added ObjectDetection App. Plus some cleanup in common files, and slightly more generic API for postModelLoad.

Mainly just refactored internal code to use Executor. The printing code was written by @ksaurabh-cadence 
Documentation:

[Optional Fixes #issue]

Test Plan:

Please see a detailed explanation of how to fill out the fields in the relevant sections in PULL_REQUEST.md.

### CODE REVIEW

**Comment by ayermolo** [comment]
@jfix71 can you take a look when you get a chance? Thanks.

File: tools/loader/ObjectDetector.cpp [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+
+/// Dumps txt files for each output and for each file.
+static int
+processAndPrintResults(const llvm::StringMap<Placeholder *> &PHM,
+                       PlaceholderBindings &bindings,
+                       llvm::ArrayRef<std::string> inputImageBatchFilenames) {
+  // Print out the object detection results.
+  std::vector<Tensor *> vecOutTensors;
+  std::vector<std::string> VecOutNames;
+  for (auto const &OutEntry : PHM) {
```
Comment (jfix71) [inline tools/loader/ObjectDetector.cpp:36]: nit: `outEntry`

File: tools/loader/ObjectDetector.cpp [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+using namespace glow;
+
+/// Dumps txt files for each output and for each file.
+static int
+processAndPrintResults(const llvm::StringMap<Placeholder *> &PHM,
+                       PlaceholderBindings &bindings,
+                       llvm::ArrayRef<std::string> inputImageBatchFilenames) {
+  // Print out the object detection results.
+  std::vector<Tensor *> vecOutTensors;
+  std::vector<std::string> VecOutNames;
```
Comment (jfix71) [inline tools/loader/ObjectDetector.cpp:35]: nit: `vecOutNames`

File: tools/loader/ObjectDetector.cpp [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+  for (auto const &OutEntry : PHM) {
+    VecOutNames.push_back(OutEntry.getKey().str());
+    Placeholder *SMPH = OutEntry.getValue();
+    Tensor *SMT = bindings.get(SMPH);
+    vecOutTensors.push_back(SMT);
+  }
+  for (unsigned i = 0; i < inputImageBatchFilenames.size(); i++) {
```
Comment (jfix71) [inline tools/loader/ObjectDetector.cpp:42]:
```suggestion
  for (unsigned i = 0, e = inputImageBatchFilenames.size(); i < e; i++) {
```

**Comment by ayermolo** [comment]
@jfix71 changed, thanks for reviewing.

**Pull request closed without merging**